src/mainwindow.cpp

```diff
@@ -100,9 +125,44 @@ void MainWindow::updateWindow(QMdiSubWindow *window)
     setWindowTitle(title);
 }
 
+void MainWindow::toolChanged(QAction *action)
+{
+    QMdiSubWindow *currentWindow = ui->mdiArea->activeSubWindow();
+    if(currentWindow != NULL){
+        PaintWidget* paintWidget = qobject_cast<PaintWidget*> (currentWindow->widget());
+        paintWidget->toolChanged(action);
+    }
+}
+
+void MainWindow::onSelectionChange()
+{
+    QMdiSubWindow *currentWindow = ui->mdiArea->activeSubWindow();
+
+    if(currentWindow != NULL){
+        PaintWidget* paintWidget = qobject_cast<PaintWidget*> (currentWindow->widget());
+        QList<Layer> items = paintWidget->getItems();
+        QList<QListWidgetItem*> selectedItems = ui->layerView->selectedItems();
+        QList<QListWidgetItem*>::iterator itrSelect = selectedItems.begin();
+        QList<Layer>::iterator itrPresent = items.begin();
+        QList<int> selected;
+        for(int i=0;itrSelect!= selectedItems.end();++itrSelect,i++){
+            for(;itrPresent != items.end();++itrPresent) {
+                if(*itrSelect == itrPresent->getListItem()){
+                    selected.push_back(i);
```

Comment:
We are too many levels deep, could be more readable if you put that in the whole part inside ` if(currentWindow != NULL)`. @hellozee

Reply:
Actually this is one of the primary reasons I have to rewrite the whole layer stuff, rewriting it would able to implement the stuff in a single for loop